llvm/lib/Transforms/Utils/Local.cpp

```diff
@@ -3311,7 +3318,10 @@ static void combineMetadata(Instruction *K, const Instruction *J,
     // TODO: Assert that this switch is exhaustive for fixed MD kinds.
     switch (Kind) {
       default:
-        K->setMetadata(Kind, nullptr); // Remove unknown metadata
+        if (K->isAtomic() && IsAMDGPUMD(Kind))
+          K->setMetadata(Kind, MDNode::intersect(JMD, KMD));
```

Comment:
It looks like all of this metadata accepts an empty node? Using intersect for this doesn't really make sense.